components/nexusoperations/executors.go

```diff
@@ -205,10 +204,35 @@ func (e taskExecutor) executeInvocationTask(ctx context.Context, env hsm.Environ
 		return fmt.Errorf("%w: %w", queues.NewUnprocessableTaskError("failed to generate a callback token"), err)
 	}
 
-	callCtx, cancel := context.WithTimeout(
-		ctx,
-		e.Config.RequestTimeout(ns.Name().String(), task.EndpointName),
-	)
+	// The following logic handles the operation ScheduleToCloseTimout parameter and Operation-Timeout header.
+	// The minimum of the resolved operation timeout and the configured request timeout is used for the context timeout
+	// when making the StartOperation request, according to the following logic:
+	// (ScheduleToClose set, Operation-Timeout set) -> no changes, use ScheduleToClose
+	// (ScheduleToClose set, Operation-Timeout unset) -> set Operation-Timeout to ScheduleToClose, use ScheduleToClose
+	// (ScheduleToClose unset, Operation-Timeout set) -> no changes, use Operation-Timeout
+	header := nexus.Header(args.header)
+	opTimeout := args.scheduleToCloseTimeout
+	opTimeoutHeader, set := header["Operation-Timeout"]
+	if !set && args.scheduleToCloseTimeout > 0 {
+		if header == nil {
+			header = make(nexus.Header, 1)
+		}
+		header["Operation-Timeout"] = args.scheduleToCloseTimeout.String()
+	} else if set && args.scheduleToCloseTimeout == 0 {
```

Comment:
Let's just ignore the user header and not enforce this. It's not technically supported and doesn't affect what the caller machinery scheduleToClose timeout.